Bugfix: Add used field to MD5 hash state

jakobnissen opened the pull request:
MD5.jl currently uses undocumented and unexported internals of the SHA stdlib. PR #86 to SHA.jl added a new field, `used` to the SHA states. Since MD5 uses internal SHA functions which now look for this field, MD5 tests began to fail.

This PR adds the needed `used` field, so MD5 now runs. The real fix is to not rely on SHA internals.

Closes #11

jw3126 commented:
ugh, looks like we don't have CI on this package :smile: could you add it in this PR also  :sweat_smile: ?

jw3126 commented on .github/workflows/UnitTests.yml:
```diff
@@ -0,0 +1,34 @@
+      fail-fast: false
+      matrix:
+        julia-version:
+          - '0.7' # LTS, and `BioSequences` minimally requires Julia v1.6 for `sum` with keyword init.
+          - '1'
```
```suggestion
          - '1.8'
          - '1'
```

jw3126 replied:
ah sorry you got nightly covered below

jw3126 replied:
for compat with https://github.com/JuliaCrypto/MD5.jl/pull/14

jakobnissen commented:
I just bumped the micro version, since changing compat is not a breaking change (anyone using an unsupported version will simply not be given this new version)

jakobnissen committed:
Bugfix: Add used field to MD5 hash state

MD5.jl currently uses undocumented and unexported internals of the SHA stdlib.
PR #86 to SHA.jl added a new field, `used` to the SHA states. Since MD5 uses
internal SHA functions which now look for this field, MD5 tests began to fail.

This PR adds the needed `used` field, so MD5 now runs. The real fix is to not
rely on SHA internals.

jakobnissen committed:
Add unit tests CI

jw3126 commented:
awesome thanks a lot @jakobnissen

bensetterholm commented:
Would it be okay to set the new tagged version of this library with the bugfix as an official release (so that it can be simply installed with `add MD5` in the Julia v1.11 package manager)?

jw3126 commented:
https://github.com/JuliaCrypto/MD5.jl/commit/c7927e301dd9d7c8710617d31cecd2095e86dd92#commitcomment-130204864
